test/hotspot/jtreg/runtime/Thread/TestBreakSignalThreadDump.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2022, Google and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test id=default
+ * @bug 8292695
+ * @summary Check that Ctrl-\ or Ctrl-Break (on Windows) causes HotSpot VM to print a full thread dump.
+ * @library /vmTestbase
+ *          /test/lib
+ * @run driver TestBreakSignalThreadDump
+ */
+
+/*
+ * @test id=with_jsig
+ * @bug 8292695
+ * @summary Check that Ctrl-\ causes HotSpot VM to print a full thread dump when signal chaining is used.
+ * @requires os.family == "linux" | os.family == "mac"
+ * @library /vmTestbase
+ *          /test/lib
+ * @run driver TestBreakSignalThreadDump load_libjsig
+ */
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Map;
+import jdk.test.lib.Platform;
+import jdk.test.lib.Utils;
+import jdk.test.lib.process.ProcessTools;
+import jdk.test.lib.process.OutputAnalyzer;
+import vm.share.ProcessUtils;
```

Comment:
There are a few serviceability dcmd jtreg tests that already depend on /vmTestbase:
- test/hotspot/jtreg/serviceability/dcmd/framework/HelpTest.java
- test/hotspot/jtreg/serviceability/dcmd/framework/VMVersionTest.java
- test/hotspot/jtreg/serviceability/dcmd/framework/Invalidcommandtest.java

Is there a real downside to depend on /vmTestbase?

Agree that it is a good suggestion to add a parameter to ProcessTools.sendCtrlBreak() and move it to test/lib. This function is only used in one other place datadumpreq001.java, so it's probably easy to change.
However, it's probably better to make a separate test-only RFE to move this function and add a test for JDK-8279124 together.

Reply:
make sense.

Reply:
There can be issues using vmTestbase as a library, simply due to the fact that jtreg does not handle libraries particularly well and we can end up with NoClassDefFoundError at runtime due to "split packages".

Reply:
Should I implement sendCtrlBreak() outside of /vmTestbase in this PR then?
The new test did run and pass on Github pre-submit check.

Reply:
I'm not sure we need yet-another-signalling-mechanism for tests. Did you look at  ./hotspot/jtreg/runtime/signal/ ? I'm unclear why TestSigquit is not failing due to this issue?

Reply:
Yes, I saw the tests under hotspot/jtreg/runtime/signal. TestSigquit is not failing due to the `-Xrs` flag, which prevents the JVM from installing any signal handler for SIGQUIT, so that the test can still install its own handler.

I didn't find any test to check that SIGQUIT would print a thread dump on stdout. It is good to have a test for this functionality.

Reply:
Okay stick with using the vmTestbase utilities then; if they cause a problem we can address it then. The split library problem tends to show up only on very specific testing combinations.